internal/hcp/registry/types.bucket.go

```diff
@@ -599,6 +603,32 @@ type NotAHCPArtifactError struct {
 	error
 }
 
+// AddMetadataToBuild adds metadata to a build in the HCP Packer registry.
+func (bucket *Bucket) AddMetadataToBuild(
+	ctx context.Context, buildName string, metadata packer.BuildMetadata,
+) error {
+	buildToUpdate, err := bucket.Version.Build(buildName)
+	if err != nil {
+		return err
+	}
+
+	packerMetadata := make(map[string]interface{})
+	packerMetadata["version"] = metadata.PackerVersion
+
+	var pluginsMetadata []map[string]interface{}
+	for _, plugin := range metadata.Plugins {
+		pluginMetadata := map[string]interface{}{
+			"version": plugin.Description.Version,
+			"name":    plugin.Name,
+		}
+		pluginsMetadata = append(pluginsMetadata, pluginMetadata)
+	}
+	packerMetadata["plugins"] = pluginsMetadata
```

Comment:
What do you think of moving this to BuildMetadata to a method like `metadata.GetPackerMetadata()` that returns already a `map[string]interface{}`? 
Returning a map of metadata is the BuildMetadata responsibility other than the Bucket. In the next phases, will also be more organized and easier to review if this code looks something like 

```
buildToUpdate.Metadata.Packer = metadata.GetPackerMetadata()
buildToUpdate.Metadata.CICD = metadata.GetCICDMetadata()
```

and each method responsible for writing the map

Reply:
The idea here is `packer` package will give the raw data and `hcp` package will format this as needed.

Reply:
Got it! I had a different point of view, like `hcp` doesn't care about what is inside the metadata and it is a free field, and it is up to packer writing this data with whatever format and info

Reply:
I agree with your suggestion though! Added that!